### PR TITLE
InfluxDB: Add configuration option for enabling insecure gRPC connections

### LINKED ISSUE #83732: InfluxDB-SQL: Allow Insecure gRPC
### What happened?

We run InfluxDB v3 Clustered and Grafana within the same Kubernetes cluster. We also terminate TLS at the load balancer rather than at the app.

We want Grafana to communicate with InfluxDB from within the cluster via the `svc.cluster.local` address rather than via the ingress address, which means the connection is insecure. This works with the InfluxQL query language.

When configuring the InfluxDB plugin with SQL as the query language and using the cluster-internal DNS name we get the following error:

```
ERROR: flightsql: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: tls: first record does not look like a TLS handshake"
```

Enabling `Skip TLS Verify` does not resolve this, it appears gRPC TLS verification is hardcoded to `true`: https://github.com/grafana/grafana/blob/2c2ac096fe66df155b68f1578bfc1137e7af53e8/pkg/tsdb/influxdb/influxdb.go#L85

### What did you expect to happen?

`Skip TLS Verify` would ideally allow us to disable `SecureGrpc`. 

### Did this work before?

InfluxDB SQL support is new.

### How do we reproduce it?

1. 
2.
3.

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 10.3.3
OS: MacOS
Browser: Chrome Version 122.0.6261.69

### Grafana platform?

Kubernetes

### Datasource(s)?

InfluxDB

### PR DESCRIPTION
**Pull request by jmickey**
**What is this feature?**

Adds an option in the `SQL` query language configuration for InfluxDB to skip TLS security on gRPC connections for Flight SQL.

**Why do we need this feature?**

InfluxData offers a Kubernetes deployment of V3 called "Clustered", when deployed alongside Grafana and using AWS LoadBalancers to terminate TLS attempting to connect to the cluster service endpoint for InfluxDB fails due to a lack of TLS connection when using the "SQL" Query Language type.

**Who is this feature for?**

InfluxDB v3 Clustered users.

**Which issue(s) does this PR fix?**:

Fixes #83732

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by jdbaldry — COMMENTED** [review]
Docs look fine to me.

File: public/app/plugins/datasource/influxdb/components/editor/config/InfluxSQLConfig.tsx [hunk 1]
```diff
@@ -57,6 +57,21 @@ export const InfluxSqlConfig = (props: Props) => {
           isConfigured={Boolean(secureJsonFields && secureJsonFields.token)}
         />
       </Field>
+      <Field horizontal label={<InlineLabel width={WIDTH_SHORT}>Insecure gRPC</InlineLabel>} className={styles.horizontalField}>
```
Comment (itsmylife) [inline public/app/plugins/datasource/influxdb/components/editor/config/InfluxSQLConfig.tsx:60]:
```suggestion
      <Field horizontal label={<InlineLabel width={WIDTH_SHORT}>Insecure Connection</InlineLabel>} className={styles.horizontalField}>
```

File: public/app/plugins/datasource/influxdb/components/editor/config/InfluxSQLConfig.tsx [hunk 2]
```diff
@@ -57,6 +57,21 @@ export const InfluxSqlConfig = (props: Props) => {
...
+        <InlineSwitch
+          id={`${htmlPrefix}-insecure-grpc`}
+          value={jsonData.insecureGrpc}
```
Comment (itsmylife) [inline public/app/plugins/datasource/influxdb/components/editor/config/InfluxSQLConfig.tsx:63]:
```suggestion
          value={jsonData.insecureGrpc ?? false}
```

**Review by itsmylife — CHANGES_REQUESTED** [review]
Thanks for the PR. It looks great. 
Just a few nits. 
After that it is good to go.

File: docs/sources/datasources/influxdb/_index.md [hunk 3]
```diff
@@ -130,6 +130,7 @@ Configure these options if you select the SQL query language:
 | ------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | **Database** | Sets the ID of the bucket to query. Copy this from the Buckets page of the InfluxDB UI.                                                                                                                                       |
 | **Token**    | API token used for SQL queries. It can be generated on InfluxDB Cloud dashboard under [Load Data > API Tokens](https://docs.influxdata.com/influxdb/cloud-serverless/get-started/setup/#create-an-all-access-api-token) menu. |
+| **Insecure gRPC** | Disable gRPC TLS security.                                                                                                                                 |
```
Comment (itsmylife) [inline docs/sources/datasources/influxdb/_index.md:133]:
```suggestion
| **Insecure Connection** | Disable gRPC TLS security.                                                                                                                                 |
```

**Comment by jmickey** [comment]
@itsmylife thanks for the review! I've applied the suggestions you made. Much appreciated

**Comment by itsmylife** [comment]
Could you please run 

- `yarn run prettier:checkDocs` 
- `yarn run prettier:write`

and commit the changes you'll get?

**Comment by jmickey** [comment]
Done!

**Comment by itsmylife** [comment]
There was a problem in `main`. That was causing one test to fail always. We've fixed it. Could you please pull the latest `main` and merge into your branch? And then hopefully I will be able to merge this one. Thanks for that tin advance!

**Review by itsmylife — APPROVED** [review]

**Comment by jmickey** [comment]
No problem! Rebased now.

**Comment by hiltontj** [comment]
It is not clear to me whether this was released with `11.0.0`. I do _not_ see it on the [release page](https://github.com/grafana/grafana/releases/tag/v11.0.0) for the `v11.0.0` tag, but I do see the changes on the version associated with the `v11.0.0` tag.

**Comment by itsmylife** [comment]
@hiltontj This is released in `v11.0.0` but somehow in the release page it wasn't mentioned. I am not sure what happened there bu this is in `v11.0.0`. Let me know if there is something else and I'd be happy to help.

**Pull request merged**